### PR TITLE
Dev refactor gmock imu interface

### PR DESCRIPTION
**Pull request by ThorsHamster**
This pullrequest is used to prepare for further work on the IMU and to update the IMU components.

The main goal was to use gMock for the IMU interface as well. Up to now the stubs were also used for the main application. This was already changed to a skeleton for future development. To prevent the codecoverage from slipping, pseudotests have been added. These will be filled via TDD in the further development.
To make the interface of the IMU testable, it was necessary to implant a Seam of Dependencies (UnitTestSetImuSeam).
Furthermore the names were changed from MPU9250 to MPU9255.
Additionally I added static casts in SetBit and ClearBit to remove the compiler warnings.

### CODE REVIEW

**Comment by elheck** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- src/imu/mpu9255.hpp  1
- src/imu/generic_imu.hpp  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/SquidSquad/Squiddrone/pullRequest?prid=6093276&bid=19950479)

File: tests/mock_libraries/imu/mock_i2c.hpp [hunk 1]
```diff
@@ -11,3 +14,5 @@ class MOCKI2C : public I2CInterface {
   MOCK_METHOD(types::DriverStatus, Write, (std::uint8_t address, const std::vector<std::uint8_t>& data, std::uint32_t timeout), (noexcept));
```
Comment (elheck) [inline tests/mock_libraries/imu/mock_i2c.hpp:14]: Why is Mock class all CAPS?

File: tests/mock_libraries/imu/mock_mpu9255.hpp [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+#define MOCK_MPU9255_HPP_
+
+#include <gmock/gmock.h>
+#include <memory>
+#include "generic_imu.hpp"
+#include "mock_i2c.hpp"
+
+namespace imu {
+
+class MOCKMPU9255 : public GenericInertialMeasurementUnit {
```
Comment (elheck) [inline tests/mock_libraries/imu/mock_mpu9255.hpp]: Why all caps?

File: src/imu/imu_interface.hpp [hunk 3]
```diff
@@ -87,6 +87,8 @@ class InertialMeasurementInterface {
    */
   virtual auto GetTemperature(void) noexcept -> int = 0;
 
+  virtual auto SetImuSeam(std::unique_ptr<imu::GenericInertialMeasurementUnit> imu) noexcept -> void = 0;
```
Comment (elheck) [inline src/imu/imu_interface.hpp]: No documentation although it is part of the public interface

**Review by elheck — CHANGES_REQUESTED** [review]

File: tests/mock_libraries/imu/mock_mpu9255.hpp (hunk 2 above)
Comment (ThorsHamster) [reply tests/mock_libraries/imu/mock_mpu9255.hpp]: Good question. Fixed here: https://github.com/Squiddrone/Squiddrone/pull/269/commits/efd5497380aa3f8abe2e8d2fdfa06c2cccfdcff4

File: tests/mock_libraries/imu/mock_i2c.hpp (hunk 1 above)
Comment (ThorsHamster) [reply tests/mock_libraries/imu/mock_i2c.hpp:14]: Good question. Fixed here: https://github.com/Squiddrone/Squiddrone/pull/269/commits/5d0fcb5c8f5199d935733c0de6c800b7b0092e2f

File: src/imu/imu_interface.hpp (hunk 3 above)
Comment (ThorsHamster) [reply src/imu/imu_interface.hpp]: Added documentation here: https://github.com/Squiddrone/Squiddrone/pull/269/commits/a6809cdad043473167d102e6c33a65f24ad24c2a
I think we should basically discuss whether public documentation for self-explanatory interfaces (this is not such an interface) is really necessary. Uncle Bob says it is not necessary if the name is well chosen.

**Comment by elheck** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- src/imu/generic_imu.hpp  1
- src/imu/mpu9255.hpp  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/SquidSquad/Squiddrone/pullRequest?prid=6093276&bid=19950479)

**Review by elheck — APPROVED** [review]

**Comment by elheck** [comment]
LGTM

**Pull request merged**